wrangler-core/src/main/java/io/cdap/wrangler/utils/Json2Schema.java

```diff
@@ -87,7 +87,8 @@ public Schema toSchema(String id, Row row) throws RecordConvertorException {
       }
 
       if (value instanceof BigDecimal) {
-        Schema schema = Schema.nullableOf(Schema.of(Schema.Type.DOUBLE));
+        BigDecimal decimal = (BigDecimal) value;
+        Schema schema = Schema.nullableOf(Schema.decimalOf(decimal.precision(), decimal.scale()));
```

Comment:
Decimals are one of the reasons determining schema from inspecting values is bad... along with unions. It can very easily be the case that the BigDecimal here happens to have a lower precision than some of the other values. 

I'm not sure there's a good way to handle this without re-designing wrangler... we might have to just always use a big precision number? Please open a jira to handle this better.

Reply:
Do you know what the maximum precision supported by common data storage system is? For example, BigQuery numeric types is 38. In the short term, I think we should just use the maximum that we know of.

Reply:
For dataprep filed: https://issues.cask.co/browse/CDAP-15361

Reply:
Since we are only supporting this type in bq connection, I have kept precision as 38.